### PR TITLE
fix impedance

### PR DESCRIPTION
**Pull request by AurelienJaquier**
## Description

- When stimulus starts at 0 for impedance (which can happen often), use steady_state_voltage_stimend to get holding voltage, and equivalent for holding current. Since at the end of the stimulus, we expect the current and voltage to vary rapidly around the holding value, this is a good enough proxy to get it.
- implemented test_steady_state_current_stimend, along with documentation and test
- fixed current interpolation
- fixed stim_end value in impedance test data file

## Checklist:
- [x] Unit tests are added to cover the changes (skip if not applicable).
- [x] The changes are mentioned in the documentation (skip if not applicable).
- [x] CHANGELOG file is updated (skip if not applicable).

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -5,6 +5,15 @@ All notable changes to this project will be documented in this file.
 The format is based on `Keep a Changelog <https://keepachangelog.com/en/1.0.0/>`_,
 and this project adheres to `Semantic Versioning <https://semver.org/spec/v2.0.0.html>`_.
 
+5.7.1 - 2024-07
+---------------
+
+- When stimulus starts at 0 for impedance (which can happen often), use test_steady_state_voltage_stimend to get holding voltage, and equivalent for holding current.
```
Comment (ilkilic) [inline CHANGELOG.rst:11]: I think you meant `steady_state_voltage_stimend`

Comment (AurelienJaquier) [reply CHANGELOG.rst]: done in latest commit

File: efel/cppcore/Subthreshold.cpp [hunk 2]
```diff
@@ -76,6 +76,40 @@ int Subthreshold::steady_state_voltage_stimend(mapStr2intVec& IntFeatureData,
...
+
+  double start_time = stimEnd - 0.1 * (stimEnd - stimStart);
+  auto start_it = find_if(times.begin(), times.end(),
+                          [start_time](double x) { return x >= start_time; });
+  auto stop_it = find_if(times.begin(), times.end(),
+                         [stimEnd](double x) { return x >= stimEnd; });
+  size_t start_index = distance(times.begin(), start_it);
+  size_t stop_index = distance(times.begin(), stop_it);
+
+  double mean = accumulate(currents.begin() + start_index,
```
Comment (ilkilic) [inline efel/cppcore/Subthreshold.cpp:99]: you can even compute the mean inside the if block

**Review by ilkilic — APPROVED** [review]
Looks good!

File: efel/cppcore/Subthreshold.cpp (hunk 2 above)
Comment (AurelienJaquier) [reply efel/cppcore/Subthreshold.cpp]: done in latest commit

Comment (ilkilic) [reply efel/cppcore/Subthreshold.cpp]: Formidable! Thank you

**Pull request merged**